src/datanode/src/instance/grpc.rs

```diff
@@ -51,17 +56,31 @@ impl Instance {
         self.query_engine
             .execute(&LogicalPlan::DfPlan(logical_plan))
             .await
-            .context(ExecuteSqlSnafu)
+            .context(ExecuteLogicalPlanSnafu)
     }
 
     async fn handle_query(&self, query: Query, ctx: QueryContextRef) -> Result<Output> {
-        Ok(match query {
+        match query {
             Query::Sql(sql) => {
                 let stmt = QueryLanguageParser::parse_sql(&sql).context(ExecuteSqlSnafu)?;
-                self.execute_stmt(stmt, ctx).await?
+                match stmt {
+                    // TODO(LFC): Remove SQL execution branch here.
+                    // Keep this because substrait can't handle much of SQLs now.
+                    QueryStatement::Sql(Statement::Query(_)) => {
+                        let plan = LogicalPlanner::new(self.query_engine.state())
```

Comment:
discuss: on the API design of planner, WDYT of `LogicalPlanner::new(self.query_engine.state())` VS `self.query_engine.planner()`, where `planner` returns a `Arc<dyn LogicalPlanner>`
 
Pros for the latter:

- Since planner is tightly associated with a query_engine implementation, get an abstract version from query_engine prevents it locked in with a particular impl.
- It avoids expose `state`, which is pretty much a datafusion concept

Reply:
make sense, I've changed the code, PTAL